### PR TITLE
Release the WorldBuilder object early.

### PR DESCRIPTION
**Pull request by bangerth**
This follows the same approach as taken for the initial conditions objects in #4889.

/rebuild

### CODE REVIEW

File: source/simulator/simulator_access.cc [hunk 1]
```diff
@@ -719,9 +719,44 @@ namespace aspect
...
+  template <int dim>
+  std::shared_ptr<const WorldBuilder::World>
+  SimulatorAccess<dim>::get_world_builder_pointer () const
+  {
+    Assert (simulator->world_builder.get() != nullptr,
+            ExcMessage ("You are trying to access the WorldBuilder "
+                        "object, but the Simulator object is not currently storing "
+                        "a pointer to such an object. This may be because you have "
+                        "not enabled WorldBuilder support during configuration of "
+                        "ASPECT. Alternatively, the Simulator object may no "
```
Comment (tjhei) [inline source/simulator/simulator_access.cc:749]: This implementation is inside an ``#ifdef ASPECT_WITH_WORLDBUILDER``, so I don't think this situation can happen. Would it make sense to remove this sentence?

**Review by tjhei — APPROVED** [review]
Good to go otherwise.

**Review by MFraters — APPROVED** [review]
I agree with Timo's comment that this should not happen since it is in a ifdef block. On the other hand, the get_world_builder function also had this message before, even though it is in the ifdef statement. So this would only be the reason if for some reason the ifdef is removed or not configured correctly. I am personally fine with the text staying there as it it, but also if it is removed or changed. If it it changed I think the emphasis should be more on the second part (instead of `Alternatively`, something like, `More likely`) in both texts.

**Comment by bangerth** [comment]
Yes, I copied that text. The idea was to have the documentation be independent of the `#ifdef`. I can go either way -- @tjhei your call!

**Comment by tjhei** [comment]
I don't see the danger of the ifdef "not working" or somebody moving the function out. I would remove the sentence.

**Comment by bangerth** [comment]
So updated!

**Review by tjhei — APPROVED** [review]

**Pull request merged**